src/mlpack/core/util/log.hpp

```diff
@@ -53,46 +69,98 @@ namespace mlpack {
  *
  * @see PrefixedOutStream, NullOutStream, IO
  */
-class Log
-{
- public:
-  /**
-   * Checks if the specified condition is true.
-   * If not, halts program execution and prints a custom error message.
-   * Does nothing in non-debug mode.
-   */
-  static void Assert(bool condition,
-                     const std::string& message = "Assert Failed.");
-
-  /**
-   * MLPACK_EXPORT is required for global variables, so that they are properly
-   * exported by the Windows compiler.
-   */
 
-  // We only use PrefixedOutStream if the program is compiled with debug
-  // symbols.
+/**
+ * MLPACK_EXPORT is required for global variables, so that they are properly
+ * exported by the Windows compiler.
+ */
+#if __cplusplus < 201703L
+    template<class Dummy>
+    struct Log_
```

Comment:
Are you sure this is necessary?  I believe that it should be sufficient just to declare member functions of `Log` as `inline`.  Perhaps I overlooked something?

Reply:
You can't inline variables before C++17.